aspects/aspects.go

```diff
@@ -845,14 +899,21 @@ func set(subKeys []string, index int, node map[string]json.RawMessage, value int
 		rawLevel = []byte("{}")
 	}
 
-	// TODO: this will error ungraciously if there isn't an object
-	// at this level (see the TODO in NewAspectBundle)
 	var level map[string]json.RawMessage
-	if err := jsonutil.DecodeWithNumber(bytes.NewReader(rawLevel), &level); err != nil {
-		return nil, err
+	err := jsonutil.DecodeWithNumber(bytes.NewReader(rawLevel), &level)
+	if err != nil {
+		var uerr *json.UnmarshalTypeError
+		if !errors.As(err, &uerr) {
+			return nil, err
+		}
+	}
+
+	// stored valued wasn't map but new write expects one so overwrite value
+	if level == nil {
+		level = make(map[string]json.RawMessage)
```

Comment:
when does this happen? tests don't hit this code atm

Reply:
When we overwrite a leaf node to be more nested. I added a small test for it